### ISSUE TITLE
Be able to create a spectrum from a ranges object

### ISSUE THREAD

**Issue by lpatiny** [cheminfo/nmr-displayer#155]
We should be able to create a spectrum from scratch. This could be achieved by filling a table in which we create AB, ABC, AB3, A2B3, etc system or from ranges.

This would be a new accordeon panel on the right for the simulation and would generate for a specific frequency, specific line width, peak shape, etc.

![image](https://user-images.githubusercontent.com/1484241/68526867-af961580-02e0-11ea-9b8a-7fd779ae8298.png)

You may get inspired by:
https://github.com/NPellet/visualizer/blob/master/src/modules/types/science/spectra/nmr/webservice_nmr_spin/view.js

and mainly the simulation.SpinSystem method:
https://github.com/NPellet/visualizer/blob/master/src/modules/types/science/spectra/nmr/webservice_nmr_spin/controller.js

https://www.npmjs.com/package/nmr-simulation

**Comment by djeanner** [comment]
Hello,
I would like to advocate an approach of spectral simulation where the link with the properties of the molecules (not just a set of <sup>1</sup>H) is taken into accout. The reason is that if we want to fit parameters to spectra we may have an approach able to include the following situations:
1) Consider the 1% natural abundance in 1H spectra (carbon satelllites). This may  be "just" to avoid to take 13C satellites as impurities, but also to determine <sup>1</sup>JCH coupling, isotopic shifts which may all be interesting to extract (in particular in a world with very high-dynamic range spectrometers. We could get residual dipolar couplings from 1D <sup>1</sup>H spectrum... (instead of complex processing of 2D HSQC).
2) If you have a silicone, you have a major signal and satellites... (a bit like <sup>13</sup>C satellites -but more intense).
3) If you have a Sn, you have pairs of satellites (in carbon spectra).
4) If you have Cl, Br, also, there are (small) but sometimes usefull isotopic effects that can be used for identification - this not very common and important, but just to say that these effects are occuring with many types of common heteroatoms.
5) We want to include effects of partial deuteration.
 
My proposition is to consider the development of a general spin system generator for organic compounds. It would allow non-natural abundance, allow simulations of <sup>1</sup>H but also 13C, 19F, <sup>2</sup>H, etc... taking into account all NMR active isotopes.

In many situations, the simulation of the spectrum would require to include multiple isotopomers. Typically for a <sup>1</sup>H spectrum, take :
1) the protons only system (with <sup>31</sup>P, <sup>19</sup>F, etc) and 
2) the _n_ isotopomers with one <sup>13</sup>C for each C. 
3) the number of systems will be multiplied by 3 for each Sn found. Rules defining how the parametes are linked (not always exactly equal - remember isotopic shifts).

With this approach, we could feed the table of coupling, chemical shfits with expected values, the coupling would be assigned!!! This interesting because one could then replace these expected values with the experimental chemical shifts and coupling constants.

**Comment by djeanner** [comment]
When a set of signals are highlighted from the 1H spectrum  or one signal is enlaged before the use clicks on "spin simulation" the table could be pre-filled with the available data... 
But here again, assignment of coupling is necessary... We realy need a tool for this! See https://github.com/cheminfo/nmr-displayer/issues/156

**Comment by nes-cgn** [comment]
I agree with that. But considering our resources, I would propose to develop the 'sophisticated' version of such a simulator as a branch, and a 'rough' version for the every-day visualizer. The main intention is to (initially at least) comply with the need of an organic chemist (but leaving an option for experts, too).
